src/modules/firewall/src/lib/Firewall.h

```diff
@@ -39,32 +40,265 @@ class FirewallLog
     static OSCONFIG_LOG_HANDLE m_logHandle;
 };
 
+class StringEnum
+{
+public:
+    StringEnum() = default;
+    StringEnum(const std::string& value) : m_value(value) {}
+    virtual ~StringEnum() = default;
+
+    virtual bool IsValid() const = 0;
+
+    std::string ToString() const
+    {
+        return m_value;
+    }
+
+    bool operator==(const char* other) const
+    {
+        return m_value.compare(other) == 0;
```

Comment:
>m_value.compare(other) == 0; [](http://example.com/codeflow?start=15&length=28)

nit: add () around, here and on line 64 and similar.

You can also use cast to bool instead ==/!= 0.

return static_cast<bool>(m_value.compare(other));

etc